include/vkwrap/vkwrap/buffer.h

```diff
@@ -87,13 +87,18 @@ class Buffer : private vk::Buffer
     // TODO: make it more type-safe.
     void update( auto* raw_data, size_t n_bytes )
     {
-        uint8_t* src = reinterpret_cast<uint8_t*>( raw_data );
-        uint8_t* dst = m_mman->map( *this );
+        assert( raw_data );
 
-        std::copy( src, src + n_bytes, dst );
-        m_mman->flush( *this );
+        if ( raw_data != nullptr )
+        {
+            uint8_t* src = reinterpret_cast<uint8_t*>( raw_data );
+            uint8_t* dst = m_mman->map( *this );
+
+            std::copy( src, src + n_bytes, dst );
+            m_mman->flush( *this );
 
-        m_mman->unmap( *this );
+            m_mman->unmap( *this );
+        }
```

Comment:
Why not pass by reference? Is nullptr a valid value?

Reply:
Probably, it's better to be reference.